module-rt/src/bootstrap.rs

```diff
@@ -43,35 +44,53 @@ impl ExportingServicePool {
     }
 }
 
+/// This service trait is compatible with FoundryModule, but slightly changed
+/// to share the Port
+#[remote_trait_object_macro::service]
+trait FoundryModuleInternal: Service {
+    fn initialize(&mut self, arg: &[u8], exports: &[(String, Vec<u8>)]);
+    fn create_port(&mut self, name: &str) -> SRwLock<dyn Port>;
+    fn debug(&mut self, arg: &[u8]) -> Vec<u8>;
+    fn shutdown(&mut self);
+}
+
```

Comment:
How "FoundryModuleInternal" is related to the "Fix termination logic" commit?

Reply:
It allows `ModuleContext` to hold `ports`.

Reply:
ok